packages/astro/src/content/utils.ts

```diff
@@ -130,10 +130,12 @@ export async function getEntryData(
 					object[schemaName] = z.preprocess(
 						async (value: unknown) => {
 							if (!value || typeof value !== 'string') return value;
-							return (await resolver(value))?.id;
+							return (
+								(await resolver(value))?.id ?? path.join(path.dirname(entry._internal.filePath))
+							);
 						},
 						schema,
-						{ description: undefined }
+						{ description: '__image' }
```

Comment:
Unrelated bug fix. Since we resetted the `description`, the paths would always be relative to the first collection loaded instead of the one currently being rendered